bundler/lib/dependabot/bundler/update_checker/latest_version_finder.rb

```diff
@@ -55,9 +57,10 @@ def fetch_lowest_security_fix_version
 
           relevant_versions = dependency_source.versions
           relevant_versions = filter_prerelease_versions(relevant_versions)
-          relevant_versions = filter_vulnerable_versions(relevant_versions)
           relevant_versions = filter_ignored_versions(relevant_versions)
           relevant_versions = filter_lower_versions(relevant_versions)
+          relevant_versions = Dependabot::UpdateCheckers::VersionFilters.filter_vulnerable_versions(relevant_versions,
+                                                                                                    security_advisories)
```

Comment:
You'll want to filtering vulnerable versions before filtering ignored versions as this code expects the set to be empty if all versions have been filtered out.

```suggestion
          relevant_versions = Dependabot::UpdateCheckers::VersionFilters.filter_vulnerable_versions(relevant_versions,
                                                                                                    security_advisories)
          relevant_versions = filter_ignored_versions(relevant_versions)
          relevant_versions = filter_lower_versions(relevant_versions)
```

Reply:
This is probably related to the npm spec failure too

Reply:
Ref: https://github.com/dependabot/dependabot-core/blob/4de04ae1967590b34f611192f4c788d6f834f7f3/bundler/lib/dependabot/bundler/update_checker/latest_version_finder.rb#L77

Reply:
Nice spot! 
@xlgmokha and I considered this fix but we preferred a fix that wasn't order-dependent. We pulled out the `Dependabot::AllVersionsIgnored` error from `#filter_ignored_versions` [into its own method](https://github.com/dependabot/dependabot-core/pull/3905/commits/1606f2c5d97551ac6e57cb5cee6d3acd4df6c998#diff-fa6fc731ba81068ee1096165004258eb11a1ff282b214ff79622dbd9d18f0ef2R89). 

We called the new error raising method in [#latest_version_from_registry](https://github.com/dependabot/dependabot-core/pull/3905/commits/1606f2c5d97551ac6e57cb5cee6d3acd4df6c998#diff-fa6fc731ba81068ee1096165004258eb11a1ff282b214ff79622dbd9d18f0ef2R45) and [#lowest_security_fix_version](https://github.com/dependabot/dependabot-core/pull/3905/commits/1606f2c5d97551ac6e57cb5cee6d3acd4df6c998#diff-fa6fc731ba81068ee1096165004258eb11a1ff282b214ff79622dbd9d18f0ef2R81)